riot/riot.py

```diff
@@ -293,10 +296,15 @@ def run(
         if any(True for r in results if r.code != 0):
             sys.exit(1)
 
-    def list_venvs(self, pattern, pythons=None, out=sys.stdout):
+    def list_venvs(self, pattern, venv_pattern, pythons=None, out=sys.stdout):
         for inst in self.venv.instances(pattern=pattern):
             if pythons and inst.py not in pythons:
                 continue
+            base_venv = get_base_venv_path(inst.py)
+            if not venv_pattern.search(
+                get_venv_directory_name(base_venv, dict(inst.pkgs))
```

Comment:
```suggestion
                get_venv_directory_name(base_venv, inst.pkgs)
```

Or was there a reason to make a copy?

Reply:
Problem here is with how this function is called in `Session.run` where `pkgs` is a dict rather than a tuple list. We can just adopt the dict for this in `VenvInstance` as well.

Reply:
Oh I see. Right, because we strip out `None` packages.. I forgot about that functionality.

Reply:
Fine to leave it as-is then IMO